src/Microsoft.ML.AutoML/AutoMLExperiment/TrialRunner.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.ML.AutoML
+{
+    internal interface ITrialRunner
+    {
+        TrialResult Run(TrialSettings settings);
+    }
+
+    internal class BinaryClassificationCVRunner : ITrialRunner
```

Comment:
>BinaryClassificationCVRunner : ITrialRunner [](http://example.com/codeflow?start=19&length=43)

I feel like these classes should be refactored. It seems like 90% of the code here is the same, couldn't you change ITrialRunner into an abstract base class that does all the shared work and the actual implementations only have to do the remaining small amount?

Reply:
I hardly think so. Although 90% of the code looks the same, actually it's not. The code for training a model, evaluating a model and creating training result is different for each scenario and each dataset setting, so the benefits of pulling out all trial runners's shared code into a base class is small considering that the training/evaluation/creating result code still need to be somewhere. In this case I'd prefer to not having a common class for those trial runners in this case.